WcaOnRails/app/models/competition.rb

```diff
@@ -216,7 +227,7 @@ def end_date
   end
 
   private def dates_must_be_valid
-    if self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
+    if !isConfirmed && self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
```

Comment:
Hmm... I don't see how adding `if: :isConfirmed?` to the `validate :dates_must_be_valid` rule would give us the same behavior. We still want to run this validation even when the competition is not confirmed.

I could add a `dates_must_be_present` validation that only runs `if: :isConfirmed?`. Does that sound better?